Fix code quality checks for 2.5.0

Ошибки касаются только метаинформации, сборка проходит с -DskipChecks=true. Однако данные нестыковки могут привести к скрытым проблемам, которые будет тяжело отладить.
`[INFO] Individual report appended to summary report.
[ERROR] Code Analysis Tool has found:
 18 error(s)!
 12 warning(s)
 95 info(s)
[ERROR] .binding.megad/build.properties:[0]
about.html file must be added to the bin.includes property
[WARNING] .binding.megad/ESH-INF/binding/binding.xml:[3]
There were whitespace characters used for indentation. Please use tab characters instead
[WARNING] .binding.megad/ESH-INF/thing/thing-types.xml:[163]
There were whitespace characters used for indentation. Please use tab characters instead
[ERROR] .binding.megad/META-INF/MANIFEST.MF:[0]
The package org.openhab.binding.megad.i2c should be marked as "internal" if it is not exported.
[WARNING] .binding.megad/META-INF/MANIFEST.MF:[7]
Unexpected "JavaSE-1.7", only allowed options: [JavaSE-1.8]
[WARNING] .binding.megad/META-INF/MANIFEST.MF:[9]
Every bundle should have optional Import-Package dependency to org.eclipse.jdt.annotation.
[WARNING] .binding.megad/META-INF/MANIFEST.MF:[10]
The version of the package org.apache.commons.codec.binary should not be specified
[WARNING] .binding.megad/META-INF/MANIFEST.MF:[11]
The version of the package org.apache.commons.lang should not be specified
[WARNING] .binding.megad/META-INF/MANIFEST.MF:[19]
The exported package `org.openhab.binding.megad.handler` is not imported
[WARNING] .binding.megad/META-INF/MANIFEST.MF:[19]
The exported package `org.openhab.binding.megad` is not imported
[WARNING] .binding.megad/OSGI-INF/MegaDHandlerFactory.xml:[4]
There were whitespace characters used for indentation. Please use tab characters instead
[ERROR] org.openhab.binding.megad.handler.MegaDBridgeHandler.java:[1]
Header is missing
[WARNING] org.openhab.binding.megad.handler.MegaDBridgeHandler.java:[146]
For periodically executed jobs that do not require a fixed rate scheduleWithFixedDelay should be preferred over scheduleAtFixedRate.
[ERROR] org.openhab.binding.megad.handler.MegaDBridgeHandler.java:[183]
Avoid printStackTrace(); use a logger call instead.
[ERROR] org.openhab.binding.megad.handler.MegaDBridgeHandler.java:[357]
Avoid printStackTrace(); use a logger call instead.
[ERROR] org.openhab.binding.megad.handler.MegaDHandler.java:[2]
Header line doesn't match pattern ^ \* Copyright \(c\) 2010-2018 by the respective copyright holders\.$
[ERROR] org.openhab.binding.megad.handler.MegaDHandler.java:[131]
Avoid printStackTrace(); use a logger call instead.
[ERROR] org.openhab.binding.megad.handler.MegaDHandler.java:[134]
Avoid printStackTrace(); use a logger call instead.
[ERROR] org.openhab.binding.megad.MegaDBindingConfiguration.java:[1]
Header is missing
[ERROR] org.openhab.binding.megad.MegaDBindingConfiguration.java:[3]
An author tag is missing
[WARNING] org.openhab.binding.megad.MegaDBindingConfiguration.java:[3]
Missing Javadoc comment.
[ERROR] org.openhab.binding.megad.i2c.I2C.java:[1]
Header is missing
[ERROR] org.openhab.binding.megad.i2c.I2C.java:[16]
An author tag is missing
[WARNING] org.openhab.binding.megad.i2c.I2C.java:[16]
Missing Javadoc comment.
[ERROR] org.openhab.binding.megad.i2c.I2C.java:[92]
Avoid printStackTrace(); use a logger call instead.
[ERROR] org.openhab.binding.megad.i2c.I2C.java:[95]
Avoid printStackTrace(); use a logger call instead.
[ERROR] org.openhab.binding.megad.internal.MegaDHandlerFactory.java:[2]
Header line doesn't match pattern ^ \* Copyright \(c\) 2010-2018 by the respective copyright holders\.$
[ERROR] org.openhab.binding.megad.MegaDBindingConstants.java:[2]
Header line doesn't match pattern ^ \* Copyright \(c\) 2010-2018 by the respective copyright holders\.$
[ERROR] .binding.megad/about.html:[0]
Missing about.html file.
[ERROR] .binding.megad/META-INF/MANIFEST.MF:[18]
A good approach is to use OSGI-INF/*.xml instead of including the services metadata files separately or using common wildcard.`

--- a/.gitignore
+++ b/.gitignore
@@ -10,3 +10,6 @@
 
 # virtual machine crash logs, see http://www.java.com/en/download/help/error_hotspot.xml
 hs_err_pid*
+
+# build directory
+target/

--- a/README.md
+++ b/README.md
@@ -1,8 +1,9 @@
+# OpenHAB 2 MegaD binding
 режимы работы: "in", "out", "dimmer", "temp", "humidity", "onewire", "adc", "at", "st", "ib", "tget", "contact", в процессе "i2c".
 
-как запустить? 
+## как запустить? 
 
-# 1) через PaperUI
+### 1) через PaperUI
 
 Configuration > System > Item Linking
 
@@ -35,7 +36,7 @@ Channels
 
 link
 
-# 2) через файлы
+### 2) через файлы
 
 .things:
 
@@ -58,7 +59,7 @@ Contact MegaDContact  "[%s]" (Temperature, GF_Corridor) { channel = "megad:devic
 
 
 Принцип такой: 
-### 1) Создаем бридж в файле .things.
+#### 1) Создаем бридж в файле .things.
 
 ```
 Bridge megad:bridge:megadeviceincoming {}
@@ -66,7 +67,7 @@ Bridge megad:bridge:megadeviceincoming {}
 
 megad:bridge: - обязятельное поле, после двоеточия - произвольное название.
 
-### 2) Добавляем Thing (По сути наши порты для меги) внутрь фигурных скобок
+#### 2) Добавляем Thing (По сути наши порты для меги) внутрь фигурных скобок
 ```
 Bridge megad:bridge:megadeviceincoming
 {
@@ -77,7 +78,7 @@ Thing device bedroomcontact [hostname="localhost", port="2", refresh="0"]
 ```
 device - обязательное поле, далее произвольное название
 
-### 3) открываем .items и создаем наши переменные.
+#### 3) открываем .items и создаем наши переменные.
 ```
 Number Temperature_GF_Corridor "Temperature [%.1f °C]" <temperature> (Temperature, GF_Corridor) { channel = "megad:device:megadeviceincoming:onewire:onewire" }
 Switch MegaDBindingThing_Input "Temperature " (Temperature, GF_Corridor) { channel = "megad:device:megadeviceincoming:kitchenout:out" } 
@@ -86,8 +87,19 @@ Contact MegaDContact "[%s]" (Temperature, GF_Corridor) { channel = "megad:device
 Последний параметр - режимы работы(каналы). до этого - путь, который мы создали в .things (megad:device:megadeviceincoming: - это название бриджа, bedroomcontact: - название Thing )
 
 
-### 4) Далее аналогично 1 версии опенхаба
+#### 4) Далее аналогично 1 версии опенхаба
 
+## Как собрать?
+
+1. Настроить IDE по [официальной инструкции](https://www.openhab.org/docs/developer/development/ide.html) и проверить что проект собирается.
+2. Скопировать директорию [org.openhab.binding.megad](org.openhab.binding.megad) в папку `<openhab2 root directory>/git/openhab2-addons/addons/binding`.
+3. Перейти в скопированную папку и выполнить `mvn clean install`. Сборка должна пройти успешно и в папке `target` появиться архив с байндингом:
+```bash
+org.openhab.binding.megad git:(master) ✗ ls -l target | grep megad
+-rw-r--r--   1 xxxxxxx  staff  29482 10 мар 21:35 org.openhab.binding.megad-2.5.0-SNAPSHOT.jar
+```
+
+## Еще
 
 По многочисленным просьбам - Donate:
 

--- a/org.openhab.binding.megad/ESH-INF/thing/thing-types.xml
+++ b/org.openhab.binding.megad/ESH-INF/thing/thing-types.xml
@@ -160,21 +160,21 @@
 		<state readOnly="true" pattern="%s"></state>
 	</channel-type>
 	<channel-type id="i2cdisplay">
-        <item-type>Text</item-type>
-        <label>i2cdisplay</label>
-        <description>i2cdisplay </description>
-        <state readOnly="true" pattern="%s"></state>
+		<item-type>Text</item-type>
+		<label>i2cdisplay</label>
+		<description>i2cdisplay </description>
+		<state readOnly="true" pattern="%s"></state>
     </channel-type>
-        <channel-type id="smsphone">
-        <item-type>String</item-type>
-        <label>sms_phone</label>
-        <description>sms_phone</description>
-        <state readOnly="true"></state>
-    </channel-type>
-        <channel-type id="smstext">
-        <item-type>String</item-type>
-        <label>sms_text</label>
-        <description>sms_text</description>
-        <state readOnly="true"></state>
-    </channel-type>
+		<channel-type id="smsphone">
+		<item-type>String</item-type>
+		<label>sms_phone</label>
+		<description>sms_phone</description>
+		<state readOnly="true"></state>
+	</channel-type>
+		<channel-type id="smstext">
+		<item-type>String</item-type>
+		<label>sms_text</label>
+		<description>sms_text</description>
+		<state readOnly="true"></state>
+	</channel-type>
 </thing:thing-descriptions>

--- a/org.openhab.binding.megad/META-INF/MANIFEST.MF
+++ b/org.openhab.binding.megad/META-INF/MANIFEST.MF
@@ -3,18 +3,20 @@ Bundle-ManifestVersion: 2
 Bundle-Name: MegaD Binding
 Bundle-SymbolicName: org.openhab.binding.megad;singleton:=true
 Bundle-Vendor: openHAB
-Bundle-Version: 2.0.4.3
-Bundle-RequiredExecutionEnvironment: JavaSE-1.7
+Bundle-Version: 2.5.0
+Bundle-RequiredExecutionEnvironment: JavaSE-1.8
 Bundle-ClassPath: .
 Import-Package: com.google.common.collect,
- org.apache.commons.codec.binary;version="1.6.0",
- org.apache.commons.lang;version="2.6.0",
+ org.apache.commons.codec.binary,
+ org.apache.commons.lang,
  org.eclipse.smarthome.config.core,
  org.eclipse.smarthome.core.library.types,
  org.eclipse.smarthome.core.thing,
  org.eclipse.smarthome.core.thing.binding,
  org.eclipse.smarthome.core.types,
- org.slf4j
-Service-Component: OSGI-INF/*
+ org.slf4j,
+ org.eclipse.jdt.annotation
+Service-Component: OSGI-INF/*.xml
 Export-Package: org.openhab.binding.megad,
- org.openhab.binding.megad.handler
+ org.openhab.binding.megad.handler,
+ org.openhab.binding.megad.i2c

--- a/org.openhab.binding.megad/OSGI-INF/MegaDHandlerFactory.xml
+++ b/org.openhab.binding.megad/OSGI-INF/MegaDHandlerFactory.xml
@@ -1,19 +1,19 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <!--
 
-    Copyright (c) 2014 openHAB UG (haftungsbeschraenkt) and others.
-    All rights reserved. This program and the accompanying materials
-    are made available under the terms of the Eclipse Public License v1.0
-    which accompanies this distribution, and is available at
-    http://www.eclipse.org/legal/epl-v10.html
+	Copyright (c) 2014 openHAB UG (haftungsbeschraenkt) and others.
+	All rights reserved. This program and the accompanying materials
+	are made available under the terms of the Eclipse Public License v1.0
+	which accompanies this distribution, and is available at
+	http://www.eclipse.org/legal/epl-v10.html
 
 -->
 <scr:component xmlns:scr="http://www.osgi.org/xmlns/scr/v1.1.0" immediate="true" name="org.openhab.binding.megad.internal.MegaDHandlerFactory">
 
-   <implementation class="org.openhab.binding.megad.internal.MegaDHandlerFactory"/>
+	<implementation class="org.openhab.binding.megad.internal.MegaDHandlerFactory"/>
 
-   <service>
+	<service>
       <provide interface="org.eclipse.smarthome.core.thing.binding.ThingHandlerFactory"/>
-   </service>
+	</service>
 
 </scr:component>

--- a/org.openhab.binding.megad/about.html
+++ b/org.openhab.binding.megad/about.html
@@ -1,0 +1,28 @@
+<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
+        "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
+<html xmlns="http://www.w3.org/1999/xhtml">
+<head>
+    <meta http-equiv="Content-Type" content="text/html; charset=ISO-8859-1"/>
+    <title>About</title>
+</head>
+<body lang="EN-US">
+<h2>About This Content</h2>
+
+<p>March 09, 2019</p>
+<h3>License</h3>
+
+<p>The openHAB community makes available all content in this plug-in (&quot;Content&quot;). Unless otherwise
+    indicated below, the Content is provided to you under the terms and conditions of the
+    Eclipse Public License Version 1.0 (&quot;EPL&quot;). A copy of the EPL is available
+    at <a href="http://www.eclipse.org/legal/epl-v10.html">http://www.eclipse.org/legal/epl-v10.html</a>.
+    For purposes of the EPL, &quot;Program&quot; will mean the Content.</p>
+
+<p>If you did not receive this Content directly from the openHAB community, the Content is
+    being redistributed by another party (&quot;Redistributor&quot;) and different terms and conditions may
+    apply to your use of any object code in the Content. Check the Redistributor's license that was
+    provided with the Content. If no such license exists, contact the Redistributor. Unless otherwise
+    indicated below, the terms and conditions of the EPL still apply to any source code in the Content
+    and such source code may be obtained at <a href="http://www.openhab.org/">openhab.org</a>.</p>
+
+</body>
+</html>

--- a/org.openhab.binding.megad/build.properties
+++ b/org.openhab.binding.megad/build.properties
@@ -3,5 +3,6 @@ output.. = target/classes
 bin.includes = META-INF/,\
                .,\
                OSGI-INF/,\
-               ESH-INF/
+               ESH-INF/,\
+               about.html
 additional.bundles = org.eclipse.jdt.annotation

--- a/org.openhab.binding.megad/pom.xml
+++ b/org.openhab.binding.megad/pom.xml
@@ -4,9 +4,9 @@
   <modelVersion>4.0.0</modelVersion>
 
   <parent>
-    <groupId>org.openhab</groupId>
-    <artifactId>binding</artifactId>
-    <version>2.0.0-SNAPSHOT</version>
+    <groupId>org.openhab.binding</groupId>
+    <artifactId>pom</artifactId>
+    <version>2.5.0-SNAPSHOT</version>
   </parent>
 
   <properties>
@@ -16,7 +16,7 @@
 
   <groupId>org.openhab.binding</groupId>
   <artifactId>org.openhab.binding.megad</artifactId>
-  <version>2.0.0-SNAPSHOT</version>
+  <version>2.5.0-SNAPSHOT</version>
   
   <name>MegaD Binding</name>
   <packaging>eclipse-plugin</packaging>

--- a/org.openhab.binding.megad/src/main/java/org/openhab/binding/megad/MegaDBindingConfiguration.java
+++ b/org.openhab.binding.megad/src/main/java/org/openhab/binding/megad/MegaDBindingConfiguration.java
@@ -1,5 +1,16 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
 package org.openhab.binding.megad;
 
+/**
+ * @author Petr Shatsillo - Initial contribution
+ */
 public class MegaDBindingConfiguration {
     public Integer port;
 }

--- a/org.openhab.binding.megad/src/main/java/org/openhab/binding/megad/MegaDBindingConstants.java
+++ b/org.openhab.binding.megad/src/main/java/org/openhab/binding/megad/MegaDBindingConstants.java
@@ -1,5 +1,6 @@
 /**
- * Copyright (c) 2014 openHAB UG (haftungsbeschraenkt) and others.
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
  * All rights reserved. This program and the accompanying materials
  * are made available under the terms of the Eclipse Public License v1.0
  * which accompanies this distribution, and is available at

--- a/org.openhab.binding.megad/src/main/java/org/openhab/binding/megad/handler/MegaDBridgeHandler.java
+++ b/org.openhab.binding.megad/src/main/java/org/openhab/binding/megad/handler/MegaDBridgeHandler.java
@@ -1,3 +1,11 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
 package org.openhab.binding.megad.handler;
 
 import java.io.BufferedReader;
@@ -180,7 +188,7 @@ public class MegaDBridgeHandler extends BaseBridgeHandler {
             this.is = s.getInputStream();
             this.os = s.getOutputStream();
         } catch (IOException e) {
-            e.printStackTrace();
+            logger.error("Couldn't start http socket.", e);
         }
 
         String input = readInput();
@@ -354,7 +362,7 @@ public class MegaDBridgeHandler extends BaseBridgeHandler {
             ss.close();
         } catch (IOException e) {
             // TODO Auto-generated catch block
-            e.printStackTrace();
+            logger.error("Couldn't dispose.", e);
         }
         updateStatus(ThingStatus.OFFLINE); // Set all State to offline
     }

--- a/org.openhab.binding.megad/src/main/java/org/openhab/binding/megad/handler/MegaDHandler.java
+++ b/org.openhab.binding.megad/src/main/java/org/openhab/binding/megad/handler/MegaDHandler.java
@@ -1,5 +1,6 @@
 /**
- * Copyright (c) 2014 openHAB UG (haftungsbeschraenkt) and others.
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
  * All rights reserved. This program and the accompanying materials
  * are made available under the terms of the Eclipse Public License v1.0
  * which accompanies this distribution, and is available at
@@ -128,11 +129,9 @@ public class MegaDHandler extends BaseThingHandler {
             }
             con.disconnect();
         } catch (MalformedURLException e) {
-            logger.error("1" + e);
-            e.printStackTrace();
+            logger.error("1" + e, e);
         } catch (ProtocolException e) {
-            logger.error("2" + e);
-            e.printStackTrace();
+            logger.error("2" + e, e);
         } catch (IOException e) {
             logger.error("Connect to megadevice " + getThing().getConfiguration().get("hostname").toString()
                     + " error: " + e.getLocalizedMessage());

--- a/org.openhab.binding.megad/src/main/java/org/openhab/binding/megad/i2c/I2C.java
+++ b/org.openhab.binding.megad/src/main/java/org/openhab/binding/megad/i2c/I2C.java
@@ -1,3 +1,11 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
 package org.openhab.binding.megad.i2c;
 
 import java.io.IOException;
@@ -13,6 +21,9 @@ import org.apache.commons.lang.ArrayUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+/**
+ * @author Petr Shatsillo - Initial contribution
+ */
 public class I2C {
 
     private Logger logger = LoggerFactory.getLogger(I2C.class);
@@ -88,11 +99,9 @@ public class I2C {
             }
             con.disconnect();
         } catch (MalformedURLException e) {
-            logger.error("1" + e);
-            e.printStackTrace();
+            logger.error("1" + e, e);
         } catch (ProtocolException e) {
-            logger.error("2" + e);
-            e.printStackTrace();
+            logger.error("2" + e, e);
         } catch (IOException e) {
             logger.error("Connect to megadevice " + host + " error: " + e.getLocalizedMessage());
         }

--- a/org.openhab.binding.megad/src/main/java/org/openhab/binding/megad/internal/MegaDHandlerFactory.java
+++ b/org.openhab.binding.megad/src/main/java/org/openhab/binding/megad/internal/MegaDHandlerFactory.java
@@ -1,5 +1,6 @@
 /**
- * Copyright (c) 2014 openHAB UG (haftungsbeschraenkt) and others.
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
  * All rights reserved. This program and the accompanying materials
  * are made available under the terms of the Eclipse Public License v1.0
  * which accompanies this distribution, and is available at

